promql/engine.go

```diff
@@ -2232,6 +2239,50 @@ func (ev *evaluator) VectorscalarBinop(op parser.ItemType, lhs Vector, rhs Scala
 	return enh.Out
 }
 
+// RangeVectorscalarbinop evaluates a binary operation between a RangeVector and a Scalar.
+func (ev *evaluator) RangeVectorscalarbinop(op parser.ItemType, lhs, rhs parser.Expr, swap bool) (Matrix, storage.Warnings) {
+	var warnings storage.Warnings
+
+	lv, lws := ev.eval(lhs)
+	warnings = append(warnings, lws...)
+	rv, rws := ev.eval(rhs)
+	// lhs always contains the Vector. If the original position was different
+	// swap for calculating the value.
+	if swap {
+		lv, rv = rv, lv
+		op = op.InverseComparisonOperator()
+	}
+	compareWith := rv.(Matrix)[0].Points[0].V
+	warnings = append(warnings, rws...)
+
+	mat := lv.(Matrix)
+	outmat := make(Matrix, 0, len(mat))
+
+	for _, ss := range mat {
+
+		ps := getPointSlice(16)
```

Comment:
Is there any significance in using the size value 16??

Reply:
Is this value 16 being used in a lot of places in the existing code? If not, then we can create a constant and use that. Makes it easier to modify the value.

Reply:
Is this value 16 being used in a lot of places in the existing code? If not, then we can create a constant and use that. Makes it easier to modify the value.